packages/vite/src/node/server/pluginContainer.ts

```diff
@@ -221,6 +221,10 @@ export async function createPluginContainer(
       if (key in info) {
         return info[key]
       }
+      // Don't throw an error when returning from an async function
+      if (key === 'then') {
+        return undefined
+      }
```

Comment:
TIL about this 😮
```js
const p = new Proxy({}, { get() { throw new Error('foo') } })

async function foo() { return p }

;(async () => {
  await foo() // error
})()
```

Reply:
Yeah that threw me for a loop initially - I believe this is due to async/await being syntactic sugar for a promise, and promises being able to return promises, so the runtime needs to "probe" for a `.then()` on the return to see whether it needs to chain the promise.